File: app/src/main/java/forpdateam/ru/forpda/api/IBaseForumPost.java
package forpdateam.ru.forpda.api;

/**
 * Created by radiationx on 27.04.17.
 */

public interface IBaseForumPost {
    int getId();

    String getDate();

    int getNumber();

    String getAvatar();

    String getNick();

    String getGroupColor();

    String getGroup();

    int getUserId();

    String getReputation();

    String getBody();

    boolean isCurator();

    boolean isOnline();

    boolean canMinusRep();

    boolean canPlusRep();

    boolean canReport();

    boolean canEdit();

    boolean canDelete();
}

File: app/src/main/java/forpdateam/ru/forpda/entity/remote/IBaseForumPost.kt
package forpdateam.ru.forpda.entity.remote

/**
 * Created by radiationx on 27.04.17.
 */

interface IBaseForumPost {
    val topicId: Int
    val forumId: Int
    val id: Int
    val date: String?
    val number: Int
    val avatar: String?
    val nick: String?
    val groupColor: String?
    val group: String?
    val userId: Int
    val reputation: String?
    val body: String?
    val isCurator: Boolean
    val isOnline: Boolean
    val canMinusRep: Boolean
    val canPlusRep: Boolean
    val canReport: Boolean
    val canEdit: Boolean
    val canDelete: Boolean
    val canQuote: Boolean
}
